src/gateway/opcodes/Identify.ts

```diff
@@ -242,25 +277,42 @@ export async function onIdentify(this: WebSocket, data: Payload) {
 			})),
 		}));
 
+	// Popultaed with users from private channels, relationships.
+	// Uses a set to dedupe for us.
+	const users: Set<PublicUser> = new Set();
+
 	// Generate dm channels from recipients list. Append recipients to `users` list
-	const channels = recipients.map((x) => {
-		const channelUsers = x.channel.recipients?.map((x) =>
-			x.user.toPublicUser(),
-		);
-		if (channelUsers) users.push(...channelUsers);
-
-		// Remove ourself from the list of other users in dm channel
-		if (x.channel.isDm()) {
-			x.channel.recipients = x.channel.recipients?.filter(
-				(y) => y.id !== this.user_id,
+	const channels = recipients
+		.filter(({ channel }) => channel.isDm())
+		.map((r) => {
+			// TODO: fix the types of Recipient
+			// Their channels are only ever private (I think) and thus are always DM channels
+			const channel = r.channel as DMChannel;
+
+			// Remove ourself from the list of other users in dm channel
+			channel.recipients = channel.recipients.filter(
+				(recipient) => recipient.user.id !== this.user_id,
```

Comment:
need to modify this to account for the cases the DM is a note to self (i.e. no other recipient than self user)